fs/lister.go

```diff
@@ -170,6 +177,7 @@ func (o *Lister) IncludeDirectory(remote string) bool {
 func (o *Lister) SetError(err error) {
 	o.mu.RLock()
```

Comment:
SInce you are writing to o.err, you need to upgrade this to o.mu.Lock() and the corresponding unlock.